libvast/src/system/posix_filesystem.cpp

```diff
@@ -23,6 +23,26 @@
 
 namespace vast::system {
 
+caf::expected<atom::done>
+posix_filesystem_state::rename_single_file(const std::filesystem::path& from,
+                                           const std::filesystem::path& to) {
```

Comment:
Doesn't rename this more than a single file? Just curious why not go with a plain filesystem rename, which should work for directories as well, no?

Reply:
I'm not sure I understand the question. Are you proposing to change the name OR the semantics of this function?

Reply:
The name only, because it actually renames directories as well.

Reply:
https://en.wikipedia.org/wiki/Everything_is_a_file